IronGram/src/main/java/com/theironyard/controllers/IronGramController.java

```diff
@@ -96,20 +100,41 @@ public Photo upload(HttpSession session, MultipartFile photo,
             throw new UserNotFoundException();
         }
 
-        File dir = new File("public/photos");
-        if(!dir.exists()){
-            dir.mkdirs();
-        }
+        if (photo.getContentType().contains(checkIsImage)) {
+            File dir = new File("public/photos");
 
-        File photoFile = File.createTempFile("photo", photo.getOriginalFilename(), dir);
-        FileOutputStream fos = new FileOutputStream(photoFile);
-        fos.write(photo.getBytes());
+            if (!dir.exists()) {
+                dir.mkdirs();
+            }
 
-        Photo p = new Photo(sender, receiver, photoFile.getName());
-        photoRepository.save(p);
+            File photoFile = File.createTempFile("photo", photo.getOriginalFilename(), dir);
+            FileOutputStream fos = new FileOutputStream(photoFile);
+            fos.write(photo.getBytes());
 
-        response.sendRedirect("/");
-        return p;
+
+            Photo p = new Photo(sender, receiver, photoFile.getName());
+
+            if (makePrivate != null){
```

Comment:
The naming here makes the code a bit confusing.  Might be easier to read if both variables were asserting the same thing instead of `makePrivate` being exactly opposite of `setPhotoPublic`